tests/dryrun_test.go

```diff
@@ -684,20 +675,12 @@ var _ = Describe("[sig-compute]Dry-Run requests", decorators.SigCompute, func()
 			Expect(err).ToNot(HaveOccurred())
 
 			By("Make a Dry-Run request to update a VM Restore")
-			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
-				restore, err = virtClient.VirtualMachineRestore(restore.Namespace).Get(context.Background(), restore.Name, metav1.GetOptions{})
-				if err != nil {
-					return err
-				}
-
-				restore.Labels = map[string]string{
-					"key": "42",
-				}
+			restore, err := virtClient.VirtualMachineRestore(restore.Namespace).Get(context.Background(), restore.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
 
-				opts := metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}}
-				_, err = virtClient.VirtualMachineRestore(restore.Namespace).Update(context.Background(), restore, opts)
-				return err
-			})
+			patch := []byte(`{"metadata":{"labels":{"key":"42"}}}`)
+			opts := metav1.PatchOptions{DryRun: []string{metav1.DryRunAll}}
+			_, err = virtClient.VirtualMachineRestore(restore.Namespace).Patch(context.Background(), restore.Name, types.MergePatchType, patch, opts)
```

Comment:
Why it's `types.MergePatchType` and not `types.JSONPatchType`?

Reply:
when the patch is simple as here, we dont need json which is for more complex cases

you can see example here that does the same
https://github.com/kubevirt/kubevirt/blob/9eee6fb8c1b9fc117f431d4a22978c18da4b7e26/pkg/controller/controller_ref_manager.go#L432

Reply:
What is simple patch and not simple one? I assume that every patch contains JSON content and should update a specific key & value.

Reply:
For example but not just, quoting the document
"For more complex usecases I’d pick JSON Patch, since it is applicable to any JSON documents (unline merge patch, which is not able to set keys to null)."

Reply:
The RFC of Json is considered more complex, there are links that shows it
and also here for example https://github.com/FusionAuth/fusionauth-issues/issues/441#issuecomment-639092237

We are already using both types on kubevirt, and this PR moves from update to patch,
if we can do it on a follow-up (and focus this PR scope to move to patch) or keep it as is i prefer please.

Reply:
even k8s docs ref to the blog i posted
"For a comparison of JSON patch and JSON merge patch, see [JSON Patch and JSON Merge Patch](https://erosb.github.io/post/json-patch-vs-merge-patch/)."

https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/
please take a look there

Reply:
I will try to explain my view on this subject, which like all these references you provided, is opinionated at some level.

- Consistency: As a project, we should strive for consistency in order to avoid surprises to code readers. I think the classic JSON patch covers a wider functionality and is explicit, therefore a better fit for choosing it for consistent usage.
- Dependency on (other) values: While it is not very frequent, we do have multiple cases where we change values based on other fields in the manifest (document). We use for that the `test` operation to assure the change is not racing with other changes. It can be seen as a light functionality of `Update`.
  - A special case of the dependency is when we want to add or remove fields from the list. While I am not 100% sure, it seems there is no mechanism in the `merge` patch to assure we avoid racing with a different change. I.e. when one calculates the list based on a given state, some other client can change the list and our "patch" may revert that change unintentionally. When using the classic `json` patch, we use the `test` operation to protect for such a scenario.

I would prefer to see a consistent usage of `patch` and in rare cases, to use the other options (mainly the strategic one). Having an explicit operation has its disadvantages, but on a project this size with so many usages, I find it safer to explicitly define the operation to assure the intention is well understood.

Reply:
The whole file uses `MergePatchType` already (8 times), does it still make sense to use `JSONPatchType` ?